tests/e2e/tests/test_rds_s3_irsa.py

```diff
@@ -0,0 +1,330 @@
+from e2e.fixtures.installation import installation, clone_upstream, ebs_addon
+from e2e.fixtures.clients import (
+    kfp_client,
+    port_forward,
+    account_id,
+    session_cookie,
+    host,
+    login,
+    password,
+    client_namespace,
+    create_k8s_admission_registration_api_client,
+)
+from e2e.utils import mysql_utils
+
+from e2e.utils.utils import kubectl_apply
+
+from e2e.utils.cloudformation_resources import (
+    create_cloudformation_fixture,
+    get_stack_outputs,
+)
+from e2e.utils.custom_resources import (
+    create_katib_experiment_from_yaml,
+    get_katib_experiment,
+    delete_katib_experiment,
+)
+
+from e2e.utils.aws.iam import IAMPolicy
+
+
+from kfp_server_api.exceptions import ApiException as KFPApiException
+from kubernetes.client.exceptions import ApiException as K8sApiException
+
+from e2e.test_methods import rds_s3
+
+
+TO_ROOT_PATH = "../../"
+
+INSTALLATION_PATH_FILE = "./resources/installation_config/rds-s3.yaml"
+RDS_S3_CLOUDFORMATION_TEMPLATE_PATH = (
+    "./resources/cloudformation-templates/rds-s3-irsa.yaml"
+)
+CUSTOM_RESOURCE_TEMPLATES_FOLDER = "./resources/custom-resource-templates"
+DISABLE_PIPELINE_CACHING_PATCH_FILE = (
+    "./resources/custom-resource-templates/patch-disable-pipeline-caching.yaml"
+)
+
+DEFAULT_USER_NAMESPACE = "kubeflow-user-example-com"
+
+@pytest.fixture(scope="class")
+def installation_path():
+    return INSTALLATION_PATH_FILE
+
+
+@pytest.fixture(scope="class")
+def cfn_stack(metadata, cluster, region, request):
+    cfn_client = get_cfn_client(region)
+    ec2_client = get_ec2_client(region)
+    stack_name = rand_name("test-e2e-rds-s3-stack-")
+
+    resp = ec2_client.describe_vpcs(
+        Filters=[
+            {
+                "Name": "tag:alpha.eksctl.io/cluster-name",
+                "Values": [cluster],
+            },
+        ],
+    )
+    vpc_id = resp["Vpcs"][0]["VpcId"]
+
+    resp = ec2_client.describe_subnets(
+        Filters=[
+            {
+                "Name": "tag:alpha.eksctl.io/cluster-name",
+                "Values": [cluster],
+            },
+            {
+                "Name": "tag:aws:cloudformation:logical-id",
+                "Values": ["SubnetPublic*"],
+            },
+        ],
+    )
+    public_subnets = [s["SubnetId"] for s in resp["Subnets"]]
+
+    resp = ec2_client.describe_instances(
+        Filters=[
+            {
+                "Name": "tag:eks:cluster-name",
+                "Values": [cluster],
+            }
+        ],
+    )
+    instances = [i for r in resp["Reservations"] for i in r["Instances"]]
+    security_groups = [i["SecurityGroups"][0]["GroupId"] for i in instances]
+
+    return create_cloudformation_fixture(
+        metadata=metadata,
+        request=request,
+        cfn_client=cfn_client,
+        template_path=RDS_S3_CLOUDFORMATION_TEMPLATE_PATH,
+        stack_name=stack_name,
+        metadata_key="test_rds_s3_cfn_stack",
```

Comment:
Any reason for the key being the same as test_rds_s3.py?

Reply:
Not really, I can specify a different keyname, but going forward we won't have the old method so this name should be agnostic to irsa vs not, but can rename it if need be.